dbt/adapters/duckdb/environments/local.py

```diff
@@ -149,7 +149,15 @@ def store_relation(self, plugin_name: str, target_config: utils.TargetConfig) ->
                     + ",".join(self._plugins.keys())
                 )
         plugin = self._plugins[plugin_name]
-        plugin.store(target_config)
+
+        handle = self.handle()
+        cursor = handle.cursor()        
+        
+        df = cursor.sql(target_config.config.model.compiled_code).arrow()
```

Comment:
We only really want to do this bit for the `external_table` materialization, right? it's overhead we don't need for the legacy `external` materialization.

Reply:
As I see the store function is called just in the case when a plugin is defined so as of now all plugins that have store functionality read the first exported file and then do that into a pandas data frame and do something with that. 
I think maybe it would make sense to refactor it as: Not export when the plugin is defined and take directly the data from duckdb connection provide it directly to the store function?

Reply:
Just a note I will try to incorporate this logic in your external materialization i just showcased it here to learn how the materialization part works and to make my life easier for now 😃